src/esp/physics/RigidObject.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "RigidObject.h"
+#include <Magnum/Math/Range.h>
+
+namespace esp {
+namespace physics {
+
+RigidObject::RigidObject(scene::SceneNode* parent)
+    : scene::SceneNode{*parent} {}
+
+bool RigidObject::initializeScene(
+    assets::PhysicsSceneAttributes& physicsSceneAttributes,
+    const std::vector<assets::CollisionMeshData>& meshGroup) {
+  if (rigidObjectType_ != NONE) {
+    LOG(ERROR) << "Cannot initialized a RigidObject more than once";
+    return false;
+  }
+
+  //! Turn on scene flag
+  rigidObjectType_ = SCENE;
+  objectMotionType_ = STATIC;
+
+  return true;
+}
+
+bool RigidObject::initializeObject(
+    assets::PhysicsObjectAttributes& physicsObjectAttributes,
+    const std::vector<assets::CollisionMeshData>& meshGroup) {
+  // TODO (JH): Handling static/kinematic object type
+  if (rigidObjectType_ != NONE) {
+    LOG(ERROR) << "Cannot initialized a RigidObject more than once";
+    return false;
+  }
+
+  //! Turn on scene flag
+  rigidObjectType_ = OBJECT;
+  // default kineamtic unless a simulator is initialized...
+  objectMotionType_ = KINEMATIC;
+
+  return true;
+}
+
+bool RigidObject::removeObject() {
+  return true;
+}
+
+bool RigidObject::isActive() {
+  // NOTE: no active objects without a physics engine... (kinematics don't
+  // count)
+  return false;
+}
+
+void RigidObject::applyForce(Magnum::Vector3& force, Magnum::Vector3& relPos) {
```

Comment:
All of these functions that don't use their arguments can be defined as `(const Magnum::Vector3&, const Magnum::Vector3&)` to deal with the compiler warning and to signify that the methods don't use that argument.